src/messages/index.ts

```diff
@@ -1,42 +1,21 @@
 import BaseApi from '../baseApi';
-import { RequestParams } from '../utils/params/requestParams';
-import { createRawApiResponse } from '../utils/response/rawApiResponse';
-import { createReadRawResponse, ReadRawResponse } from '../utils/response/readRawResponse';
-import { createResourceResponse } from '../utils/response/resourceResponse';
-import * as responseHandlers from '../utils/response/responseHandlers';
-import { Message, MessageCreate, MessageUpdate } from './types';
+import { Resources } from '../utils/response/apiResponse';
+import { required, Required } from '../utils/response/responseHandlers';
+import { MessageCreate, MessageUpdate } from './types';
 
-export class MessagesEndpoint extends BaseApi {
-  public readonly resourceName = 'messages';
+const resourceName = 'messages';
+type ResourceName = typeof resourceName;
```

Comment:
Might lead to confusion between resourceName and ResourceName (and search has to be case sensitive to distinguish between those two).